R/particle_filter.R

```diff
@@ -171,7 +171,7 @@ particle_filter <- R6::R6Class(
       }
 
       log_likelihood <- 0
-      prev_state <- model$state()
+      prev_state <- state
       for (t in seq_len(private$n_steps)) {
         model$run(steps[t, 2])
         state <- model$state()
```

Comment:
At some point we really need to hold something that indicates what state we'd want out at this point - need to decide if that belongs in dust or in mcstate